common/test/test_lint.py

```diff
@@ -1,22 +1,104 @@
+# SPDX-FileCopyrightText: © 2023 Christian BUHTZ <c.buhtz@posteo.jp>
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This file is part of the program "Back In Time" which is released under GNU
+# General Public License v2 (GPLv2).
+# See file LICENSE or go to <https://www.gnu.org/licenses/#GPL>.
+"""Tests using several linters.
+
+Instead of failing tests are skipped on machines where the linters are not
+available. As an exception only on TravisCI tests are forced to run and will
+fail if linters not available.
+"""
 import unittest
 import os
 import pathlib
 import subprocess
 import shutil
 from typing import Iterable
+try:
+    import pycodestyle
+    PYCODESTYLE_AVAILABLE = True
+except ImportError:
+    PYCODESTYLE_AVAILABLE = False
+
+BASE_REASON = ('Using package {0} is mandatory on TravisCI, on '
+               'other systems it runs only if `{0}` is available.')
 
 ON_TRAVIS = os.environ.get('TRAVIS', '') == 'true'
-PYLINT_AVIALBE = not shutil.which('pylint') is None
-PYLINT_REASON = ('Using PyLint is mandatory on TravisCI, on other systems'
-                 'it runs only if `pylint` is available.')
+TRAVIS_REASON = ('Running linter tests is mandatory on TravisCI only. On '
+                 'other machines they will run only if linters available.')
+
+PYLINT_AVAILABLE = shutil.which('pylint') is not None
+RUFF_AVAILABLE = shutil.which('ruff') is not None
+
+ANY_LINTER_AVAILABLE = any((
+    PYLINT_AVAILABLE,
+    RUFF_AVAILABLE,
+    PYCODESTYLE_AVAILABLE,
+))
 
+# Files in this lists will get the full battery of linters and rule sets.
+full_test_files = [pathlib.Path(fp) for fp in (
+    'bitbase.py',
+    'schedule.py',
+    'version.py',
+    'test/test_lint.py',
+)]
 
+# Not all linters do respect PEP8
+PEP8_MAX_LINE_LENGTH = 79
```

Comment:
Consider setting [max line length](https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/line-too-long.html) in a top-level config file for the respective linter.
For example, for pylint, you could have a config file such as: `pylintrc` at the top level containing the following:

``` ini
[MAIN]
max-line-length=79
```

Using the standard config allows the correct configuration to be used when running the linters directly.

Also, the other linters such as `ruff` [also check line length](https://docs.astral.sh/ruff/rules/line-too-long/), so you might consider disabling the line length check in all linters but one.

Reply:
That constant is for ruff because ruff do not follow PEP8 in all aspects. It checks the line length with a default value different from 79.

A global config file, a pyproject.toml, is the goal. But before having this we need to fix #1575 . Other config files different from pyproject.toml are problematic because we would need several of such files, each for every linter/tool. But I would like to have the config centralized. Having the config in the test_lint.py file is kind of centralized and of course a workaround. It is IMHO easier to manage this way until #1575 is fixed.

Reply:
Understood.